pymeasure/instruments/eurotest/eurotestHPP120256.py

```diff
@@ -0,0 +1,417 @@
+        at a determinated voltage changing rate (ramp in V/s).
+        :param voltage_rate: Is the changing rate (ramp in V/s) for the ramp setting
+        """
+        log.info(f"Executing the ramp_to_zero function with ramp: {voltage_rate} V/s.")
+
+        self.voltage_ramp = voltage_rate
+        self.voltage_setpoint = 0
+
+    def wait_for_output_voltage_reached(self, voltage_setpoint, abs_output_voltage_error=0.03,
+                                        check_period=1.0, timeout=60.0):
+        """
+        Wait until HV voltage output reaches the voltage setpoint.
+
+        Checks the voltage output every check_period seconds and raises an exception
+        if the voltage output doesn't reach the voltage setting until the timeout time.
+        :param voltage_setpoint: the voltage in kVolts setted in the HV power supply which
+        should be present at the output after some time (depends on the ramp setting).
+        :param abs_output_voltage_error: absolute error in kVolts for being considered
+        an output voltage reached.
+        :param check_period: voltage output will be measured every check_period (seconds) time.
+        :param timeout: time (seconds) give to the voltage output to reach the voltage setting.
+        :return: None
+        :raises: Exception if the voltage output can't reach the voltage setting
+        before the timeout completes (seconds).
+        """
+        log.info("Executing the wait_for_output_voltage_reached function.")
+
+        ref_time = time.time()
+        future_time = ref_time + timeout
+
+        voltage_output = self.voltage
+        voltage_output_set = math.isclose(voltage_output, voltage_setpoint, rel_tol=0.0,
+                                          abs_tol=abs_output_voltage_error)
+
+        log.debug(f"\tWaiting for voltage output set. "
+                  f"Reading output voltage every {check_period} seconds.\n"
+                  f"\tTimeout: {timeout} seconds.")
+
+        while not voltage_output_set:
+            actual_time = time.time()
+
+            time.sleep(check_period)  # wait for voltage output reaches the voltage output setting
+            voltage_output = self.voltage
+            voltage_output_set = math.isclose(voltage_output, voltage_setpoint, rel_tol=0.0,
+                                              abs_tol=abs_output_voltage_error)
+            log.debug("voltage_output_valid_range: "
+                      "[" + str(voltage_setpoint - abs_output_voltage_error) +
+                      ", " + str(voltage_setpoint + abs_output_voltage_error) + "]")
+            log.debug("voltage_output: " + str(voltage_output))
+            log.debug(f"Elapsed time: {round(actual_time - ref_time, ndigits=1)} seconds.")
+
+            if actual_time > future_time:
+                self.shutdown(200.0)  # in case the voltage were applied at the output
+                raise TimeoutError("Timeout for wait_for_output_voltage_reached function")
```

Comment:
This feels more pythonic to me, just a suggestion:
```suggestion
        def target_reached(voltage_output):
            return math.isclose(voltage_output, voltage_setpoint, rel_tol=0.0,
                                abs_tol=abs_output_voltage_error)

        log.debug(f"\tWaiting for voltage output set. "
                  f"Reading output voltage every {check_period} seconds.\n"
                  f"\tTimeout: {timeout} seconds.")

        while True:
            actual_time = time.time()
            time.sleep(check_period)  # wait for voltage output reaches the voltage output setting
            voltage_output = self.voltage
            if target_reached(voltage_output):
                break
            log.debug("voltage_output_valid_range: "
                      "[" + str(voltage_setpoint - abs_output_voltage_error) +
                      ", " + str(voltage_setpoint + abs_output_voltage_error) + "]")
            log.debug("voltage_output: " + str(voltage_output))
            log.debug(f"Elapsed time: {round(actual_time - ref_time, ndigits=1)} seconds.")

            if actual_time > future_time:
                self.shutdown(200.0)  # in case the voltage were applied at the output
                raise TimeoutError("Timeout for wait_for_output_voltage_reached function")
```

Reply:
I see, but what you think if we avoid the creation of a new function and change the if as shown next? 

```
if target_reached(voltage_output):
                break
```

to  

```
if math.isclose(voltage_output, voltage_setpoint, rel_tol=0.0,
                                abs_tol=abs_output_voltage_error):
                break
```

Thanks

Reply:
Yeah, now that you refactored the loop so you don't need to check that in two different places (like before), you don't need the extra method.